### ISSUE TITLE
Search only shows results from API reference

### ISSUE THREAD

**Issue by warpech** [handsontable/hyperformula#948]
### Description

The search function in the HyperFormula docs only shows results from the API reference.

It should be improved to also show the results from the Guides, similarly how it works in the Handsontable docs.

### Links

- https://handsontable.github.io/hyperformula/
- https://handsontable.com/docs/

### Type of issue

- [ ] Missing documentation
- [ ] Error in the documentation
- [ ] Request for a demo
- [ ] Request for a tutorial, guide etc.
- [x] Other issues, suggestions or ideas

**Comment by sequba** [comment]
Related: #878

**Comment by sequba** [comment]
I investigated this issue and here are my findings:
- our search does include results from the guides (same as Handsoncode)
- it looks for the search string only in headers (same as Handsoncode)
- the list of results is limited to 5 items (it can be configured easily)
- Handsoncode search has a custom feature of grouping the results and defining limits for each group separately: 5 items for guides and 10 for API reference (this is not as easy as setting a single config param but should be doable)

IMO we should definitely display more search results than 5. Grouping search results would also be handy in case of HyperFormula. @warpech WDYT

![image](https://user-images.githubusercontent.com/2480086/165805552-297191a7-9c92-4c94-aeeb-1837b05f9c0d.png)

![image](https://user-images.githubusercontent.com/2480086/165806680-1db107a5-fffc-49bc-a134-1ad54bc7f5e3.png)

**Comment by warpech** [comment]
Good idea! The problem is how to do it quickly. I think there might be a way.

----

As you discovered, in HyperFormula docs, we are using unmodified https://github.com/vuejs/vuepress/tree/v1.9.7/packages/@vuepress/plugin-search. 

Handsontable has basically a custom theme with a heavily modified copy of this plugin. Our customizations were implemented in https://github.com/handsontable/handsontable/pull/7626. Sample commit from this PR that adds separation between categories "Guides" and "API Reference": https://github.com/handsontable/handsontable/pull/7626/commits/644bb2fdaa562caeff2b66af44d58cc1fbe2b8fd. 

----

I think it should be possible to copy [@vuepress/plugin-search](https://github.com/vuejs/vuepress/tree/v1.9.7/packages/@vuepress/plugin-search) as a custom plugin to `docs/.vuepress/plugins`. Handsontable some other custom plugins working that way: https://github.com/handsontable/handsontable/tree/8344a56d818c209a7449a2ec1217520588e013a4/docs/.vuepress/plugins

Then, you could modify this copy of `@vuepress/plugin-search` to implement categories, using parts of the solution from Handsontable: https://github.com/handsontable/handsontable/blob/8344a56d818c209a7449a2ec1217520588e013a4/docs/.vuepress/theme/components/SearchBox.vue

**Comment by sequba** [comment]
Released with **2.0.1**
